### PR TITLE
Use a view props instead to update the currently selected formats

### PR DESCRIPTION
**Pull request by Tug**
This PR removes our existing formatting callbacks (`onActiveFormatsChange` and `onActiveFormatAttributesChange `) and uses a view props (`activeFormats`) instead to update the currently selected formats.

With https://github.com/WordPress/gutenberg/pull/12249 the formatting will be handled over to gutenberg's format-library and Aztec will be "mostly" rendering the HTML. However, we also want to be able to render formatted text without delay, thus we need Aztec to be able to edit a rich text directly without formatting changes while the text entered cycles back to RN.

FYI, this "feature" is added to the format-library PR with [this commit](https://github.com/WordPress/gutenberg/pull/12249/commits/f3188d927f50c153ecc9b3c35fe92658af85755d).

I'm not a Swift programmer so I'm interested in your feedback and really welcome contributions on this one :)

**Edit:** Added an implementation for Android but it's currently buggy

### CODE REVIEW

**Comment by koke** [comment]
We can't really remove those methods until the other PRs are merged, right?

**Comment by Tug** [comment]
Well we 'll need this merged for https://github.com/wordpress-mobile/gutenberg-mobile/pull/275 but I haven't laid out a merge plan yet.

File: ios/RNTAztecView/RCTAztecView.swift [hunk 1]
```diff
@@ -20,6 +18,14 @@ class RCTAztecView: Aztec.TextView {
             blockModel = BlockModel(tag: tag)
         }
     }
+    @objc var activeFormats: NSSet? = nil {
+        didSet {
+            let currentTypingAttributes = formattingIdentifiersForTypingAttributes()
+            for (key, value) in formatStringMap where currentTypingAttributes.contains(key) != activeFormats?.contains(value) {
```
Comment (koke) [inline ios/RNTAztecView/RCTAztecView.swift]: It's a bit hard to follow but it seems this is adding any missing format to the typing attributes, but should it also be removing anything that's currently there?

**Comment by koke** [comment]
OK, let's keep reviewing here but not merge until the other parts are ready

**Comment by hypest** [comment]
[We've folded](https://github.com/wordpress-mobile/gutenberg-mobile/pull/465) this repo into gutenberg-mobile so, if this PR is still active please, migrate it to a PR on the gutenberg-mobile repo. Thanks!
